File: Practica_11/Python/Fila_B/Pintura.py
from Obra import Obra
class Pintura(Obra):
    def __init__(self, titulo, material, a1, a2, anuncio, genero):
        super().__init__(titulo, material, a1, a2, anuncio)
        self.genero = genero

    def mostrar(self):
        super().mostrar()
        print("PINTURA")
        print("Genero:", self.genero)

    def calcular_promedio_anios_experiencia(self):
        print("PROMEDIO DE AÑOS DE EXPERIENCIA DE LOS ARTISTAS")
        a = self.a1
        b = self.a2
        promedio = (a.get_anios_experiencia() + b.get_anios_experiencia()) / 2
        print("Promedio:", promedio)

    def incrementar_precio_por_artista(self, incremento, nombre_artista):
        print("INCREMENTAMOS EL PRECIO EN X A LA PINTURA DEL ARTISTA CON NOMBRE X")

        if (self.a1.get_nombre().lower() == nombre_artista.lower() or
                self.a2.get_nombre().lower() == nombre_artista.lower()):
            nuevo_precio = self.anuncio.get_precio() + incremento
            self.anuncio.set_precio(nuevo_precio)
            print("Precio Incrementado:", self.anuncio.get_precio())
        else:
            print("No se encontro al artista con nombre:", nombre_artista)

File: Practica_11/src/Java/Fila_B/Pintura.java
package Java.Fila_B;

public class Pintura extends Obra {
	String genero;
	
	public Pintura(String titulox, String materialx, Artista a1x, Artista a2x, Anuncio bx, String generox) {
		super(titulox, materialx, a1x, a2x, bx);
		this.genero=generox;
	}
	public void mostrar() {
		super.mostrar();
		System.out.println("PINTURA");
		System.out.println("Genero: "+this.genero);
	}
	//b  Calcular el promedio de años Experiencia de los artistas de ambas pinturas 
	public void CalcularPromedioAñosExperiencia() {
		System.out.println("PROMEDIO DE AÑOS DE EXPERIENCIA DE LOS ARTISTAS");
		Artista a = a1;
		Artista b = a2;
		double promedio=0;
		promedio= ((a.getAñosExperiencia()+b.getAñosExperiencia())/2);
		System.out.println("Promedio: "+promedio);
		
	}
	//c
	public void IncrementaPrecioXArtistaNombreX(double incremento, String nombreArtista) {
	    System.out.println("INCREMENTAMOS EL PRECIO EN X A LA PINTURA DEL ARTISTA CON NOMBRE X");
	    
	    if (a1.getNombre().equalsIgnoreCase(nombreArtista) || 
	        a2.getNombre().equalsIgnoreCase(nombreArtista)) {
	        
	        b.setPrecio(b.getPrecio() + incremento);
	        System.out.println("Precio Incrementado: " + b.getPrecio());
	        
	    } else {
	        System.out.println("No se encontro al artista con nombre: " + nombreArtista);
	    }
	}
}
